Slim/Control/Queries.pm

```diff
@@ -276,6 +276,8 @@ sub albumsQuery {
 	my $libraryID     = Slim::Music::VirtualLibraries->getRealId($request->getParam('library_id'));
 	my $year          = $request->getParam('year');
 	my $sort          = $request->getParam('sort') || ($roleID ? 'artistalbum' : 'album');
+	my $work	  = $request->getParam('work_id');
+	my $composerID    = $request->getParam('composer_id');
```

Comment:
`composer_id` should not be required here as it would be the same as the combination of `artist_id` and `role_id:2`, wouldn't it?

Reply:
I was thinking ahead to the (yet to be added) search criteria on Work. Say you wanted to show albums containing Beethoven's Violin Concerto with a particular performer? (Work title, on its own, is not necessarily unique, it's qualified by composer.) Does that make sense?

Reply:
While it make sense from a technical point of view I, again, lack the understanding of how the UX for this would be. I can't imagine a browse hierarchy, or even search form to do this...

Reply:
My idea is to add a composer criteria to the advanced search, so it could be run with both artist and composer search strings.
![Screenshot from 2023-11-19 15-37-18](https://github.com/Logitech/slimserver/assets/97928054/249f6d2f-2888-4f95-9623-a8169df3946c)

Reply:
Ok, got it. But I think you're opening a can of worms... why should this be limited to Composer? Why not Conductors? Bands? Any of the other roles? You'd have to write specific solutions for all of these, and that will be a mess to get right, complete, and consistent across all of them, with all possible permutations of conductor and composer and instrumentalist etc. 

I don't know whether I should wish you good luck, or suggest to not even start 😂.

Reply:
From a musical point of view this might be different. But it's not so much different from the case where you want to find the Beatles song (album artist) where Clapton (track artist) played the guitar, is it?

Reply:
There are all sorts of search combinations which ideally could be supported, but look at it this way:

In the rock/pop would a search facility without the ability to search for "The Beatles, White Album" be OK? Because in the classical world, the ability to search for "Beethoven, Symphony No 9, Berlin Philharmonic" is just as fundamental.

Reply:
What I don't like about the idea is that this particular case will likely end in too much too specific code. You'll have to deal with multiple queries against the database as there would be multiple relations between `tracks` and `contributor_tracks` to be dealt with, with different arguments each. 

While I don't see this fit in to the `albums` query, I'm having a hard time trying to come up with a better idea. I think it's just beyond what those "simple" queries can reasonably do. Similarly to what you can do with pure SQL, and what you can't without additional "business logic". I think what you're trying to do would need to be done more on the application side than in those queries. Like I did with the releases work.

For the search we could probably improve the FTS to cover your needs. But that wouldn't help with a structured approach like a menu.

Reply:
Thanks for the feedback. If it helps conceptually, the new Works table contains composer id, so you could look at a works search as searching on that field, and not the track-contributor role of composer (though once the SQL joins are done, works.composer and contributor_track.contributor with role=2 will necessarily have the same value).

Maybe to way to present this in the Search UI is to have a 2-field Works search criteria - Work Name & Work Composer.

Anyway, I'm still struggling with incorporating the Works menu item at the head of the albums list, and am on the verge of incorporating Works into the Releases menu instead. Once I've made some substantive progress, I'll update the PR.